ros/src/common/libs/state_machine_lib/test/src/test_state_hpp.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2018 Autoware Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ros/ros.h>
+#include <gtest/gtest.h>
+
+#include "state_machine_lib/state.hpp"
+
+class TestSuite: public ::testing::Test {
+public:
+	TestSuite(){}
+	~TestSuite(){}
+};
+
+TEST(TestSuite, CheckStateConstructor){
+
+	std::string state_name = "TestState";
+	uint64_t state_id = 0;
+	state_machine::State state(state_name, state_id);
+
+	ASSERT_EQ(state.getStateID(),state_id) << "_state_id should be " << state_id;
+	ASSERT_STREQ(state.getStateName().c_str(), state_name.c_str()) << "state_name should be " << state_name;
+	ASSERT_TRUE(state.getChild() == NULL) << "child_state_ pointer should be NULL";
+	ASSERT_TRUE(state.getParent() == NULL) << "parent_state_ pointer should be NULL";
+	ASSERT_STREQ(state.getEnteredKey().c_str(), "") << "entered_key should be " << "";
+
+
+	std::ostringstream oss;
```

Comment:
This is a bit fragile, why do you need to check `std::cout`? You could create a fixture that gets updated when a state transition or whatever the state machine does.

Reply:
@esteve the tests have been updated, please have a look now. Thanks!

Reply:
Looks good. I'd go further and remove `showStateName()` from the State API, so as to separate the logic from the UI (in this case the console). It's only being used in one place https://github.com/sgermanserrano/Autoware/blob/feature/test_state_machine_lib/ros/src/common/libs/state_machine_lib/src/state_context.cpp#L142